### PR TITLE
Refactor installation script.

### PR DESCRIPTION
**Pull request by fristonio**
## Description
- Refactor check_sudo function to use $EUID to check for root user.
- Refactor db_setup function
    - update logic to check if postgres service is running or not
    - fix db_setup logic.

## Related Issue
#927 

## Reviewers
@viyatb @7a 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [x] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by fristonio** [comment]
@viyatb @7a I have made the required changes to install.sh, can you please review the PR and suggest changes. :)

File: scripts/install.sh [hunk 1]
```diff
@@ -227,6 +222,39 @@ db_setup() {
...
+                echo "${info}[+] Database $db_name already exist.${reset}"
+            else
+                # Either database does not exists or the owner of database is not $db_user
+                # Create new database $db_name with owner $db_user
+                psql postgres -c "CREATE DATABASE $db_name WITH OWNER $db_user ENCODING 'utf-8' TEMPLATE template0;"
+            fi
+            echo "${normal}"
+            write_db_settings
+            echo "${reset}"
+            #sudo su postgres -c "psql -c \"CREATE USER $db_user WITH PASSWORD '$db_pass'\""
```
Comment (viyatb) [inline scripts/install.sh]: Remove any commented code.

File: scripts/install.sh [hunk 2]
```diff
@@ -227,6 +222,39 @@ db_setup() {
...
+            #sudo su postgres -c "psql -c \"CREATE DATABASE $db_name WITH OWNER $db_user ENCODING 'utf-8' TEMPLATE template0;\""
+        elif [ "$action" = "clean" ]; then
+            psql postgres -c "DROP DATABASE $db_name"
+            psql postgres -c "DROP USER $db_user"
+            # sudo su postgres -c "psql -c \"DROP DATABASE $db_name\""
```
Comment (viyatb) [inline scripts/install.sh]: Ditto.

**Comment by fristonio** [comment]
We don't use these any function anymore in the script. Should I remove them too 
https://github.com/owtf/owtf/blob/develop/scripts/install.sh#L138

**Comment by viyatb** [comment]
I think it would more useful to modify the `db_setup` function to use those functions (making it more modular).

**Comment by fristonio** [comment]
Ok, I will have those functions modified as per our need.

**Comment by fristonio** [comment]
@viyatb can you have a look, I have removed dead code from the script.

File: scripts/install.sh [hunk 3]
```diff
@@ -173,59 +239,34 @@ EOF
...
-            echo "${info}[+] PostgreSQL server is running ${postgres_server_ip}:${postgres_server_port} :)"
+            # Create database $db_name if it does not exist.
+            if postgresql_check_db; then
+                echo "${info}[+] Database $db_name already exist."
+            else
+                # Either database does not exists or the owner of database is not $db_user
+                # Create new database $db_name with owner $db_user
+                postgresql_create_db
+            fi
+            # After the database has been set up wrtite settings to db.yaml
```
Comment (viyatb) [inline scripts/install.sh]: Typo.

File: scripts/install.sh [hunk 4]
```diff
@@ -71,8 +71,12 @@ create_directory() {
 }
 
 check_sudo() {
-    timeout 2 sudo id && sudo=1 || sudo=0
-    return $sudo
+    if [[ $EUID -eq 0 ]]; then
+        # The script is invoked with root permission
```
Comment (viyatb) [inline scripts/install.sh]: This comment is still incorrect - the function is checking whether the current user has administrative privileges.

File: scripts/install.sh [hunk 5]
```diff
@@ -173,59 +235,34 @@ EOF
...
-                echo "          sudo systemctl start postgresql"
-                exit 1
+                # Create new user $db_user with password $db_pass
+                postgresql_create_user
             fi
-        else
-            echo "${info}[+] PostgreSQL server is running ${postgres_server_ip}:${postgres_server_port} :)"
+            # Create database $db_name if it does not exist.
+            if postgresql_check_db; then
+                echo "${info}[+] Database $db_name already exist."
```
Comment (viyatb) [inline scripts/install.sh]: Forgot `${reset}` here.

File: scripts/install.sh [hunk 6]
```diff
@@ -135,26 +135,88 @@ proxy_setup() {
...
+            fi
+        else
+            echo "${info}[+] On DEBIAN based distro [i.e Kali, Ubuntu etc..]"
+            echo "          sudo service postgresql start"
+            echo "${info}[+] On RPM based distro [i.e Fedora etc..]"
+            echo "          sudo systemctl start postgresql"
+            exit 1
+        fi
+    else
+        echo "${info}[+] PostgreSQL server is running ${postgres_server_ip}:${postgres_server_port} :)"
```
Comment (viyatb) [inline scripts/install.sh]: Forgot `${reset}`.

**Comment by viyatb** [comment]
thanks @fristonio !

**Pull request merged**